services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -832,10 +843,14 @@ export class WorkflowService {
 
     const updatedWorkflow = await this.updateContextById(
       workflowId,
-      {
-        documents: [newDocument],
-      },
+      this.updateDocumentInContext(runtimeData.context, newDocument, documentsUpdateContextMethod),
       [projectId],
+      documentsUpdateContextMethod === 'director' ? 'by_index' : 'by_id',
```

Comment:
Maybe we should change `contextUpdateMethod` from `base` | `director` to `by_index` | `by_id`. Unless there are more differences beyond the plv8 merge method `by_index` could be potentially be used by other places than directors. Make sure we get this past @alonp99 though.